### PR TITLE
Add documentation for core.h in C++ tutorial (and minor fixes)

### PR DESCRIPTION
**Pull request by ischoegl**
This PR contains minor updates:

- Add documentation for `core.h` in the C++ tutorial (and "todo" items for future simplifications)
- Some formatting fixes (single backslash does not render correctly in code block, see [sphinx issue](https://github.com/sphinx-doc/sphinx/issues/8396))

### CODE REVIEW

File: pages/install/conda-install.rst [hunk 1]
```diff
@@ -260,17 +263,18 @@ programs can be compiled as follows using the ``pkg-config`` build system:
...
+   library files               path/to/conda/envs/ct-dev/Library/lib
+   C++ headers                 path/to/conda/envs/ct-dev/Library/include
+   samples                     path/to/conda/envs/ct-dev/share/cantera/samples
+   data files                  path/to/conda/envs/ct-dev/share/cantera/data
 
-The development package on Windows is experimental. While all libraries and headers are
-installed, compilation of custom programs is untested.
+The development package on Windows is experimental, with Fortran 90 currently not being
+supported. While libraries and headers are installed, compilation of custom programs is
+untested.
```
Comment (bryanwweber) [inline pages/install/conda-install.rst:277]:
```suggestion
The development package on Windows is experimental, with Fortran 90 currently not being
supported.
```
I'd suggest deleting that last sentence, it doesn't add anything to me and might scare away people who could test this for us :smile:

File: pages/tutorials/cxx-guide/headers.rst [hunk 2]
```diff
@@ -16,29 +16,35 @@ These headers are designed for use in C++ application programs, and are not
 included by the Cantera core. The headers and their functions are:
 
 
+``core.h``
+    Base classes and functions for creating
+    `Solution <{{% ct_docs doxygen/html/d5/d40/classCantera_1_1Solution.html %}}>`__
+    objects from input files, as well as associated classes defining a phase
+    *(New in Cantera 3.0)*.
```
Comment (bryanwweber) [inline pages/tutorials/cxx-guide/headers.rst:23]: Nice!

**Review by bryanwweber — COMMENTED** [review]
One minor suggestion here, and thanks for taking on these updates!

**Comment by ischoegl** [comment]
@bryanwweber ... while some issues remain in `libcantera-devel` (see Cantera/conda-recipes#37), they don't prevent the example given from working. So I think this is good to go ...

**Review by bryanwweber — APPROVED** [review]
Thanks!

**Pull request merged**